### PR TITLE
Improve WebSubHub Compiler Plugin

### PR DESCRIPTION
**Pull request by ayeshLK**
## Purpose
 `websubhub:Service` remote methods should support only `http:Headers` as a parameter to specific remote methods.
```ballerina
service /websubhub on new websubhub:Listener(9090) {

    isolated remote function onRegisterTopic(websubhub:TopicRegistration message, http:Headers headers)
                                returns websubhub:TopicRegistrationSuccess|websubhub:TopicRegistrationError|error {
        // implement logic here
    }

    isolated remote function onDeregisterTopic(websubhub:TopicDeregistration message, http:Headers headers)
                        returns websubhub:TopicDeregistrationSuccess|websubhub:TopicDeregistrationError|error {

        // implement logic here
    }

    isolated remote function onUpdateMessage(websubhub:UpdateMessage message, http:Headers headers)
               returns websubhub:Acknowledgement|websubhub:UpdateMessageError|error {
        // implement logic here
    }
    
    isolated remote function onSubscription(websubhub:Subscription msg, http:Headers headers)
                returns websubhub:SubscriptionAccepted|websubhub:BadSubscriptionError|websubhub:InternalSubscriptionError|error {
        // implement logic here
    }

    isolated remote function onSubscriptionValidation(websubhub:Subscription msg)
                returns websubhub:SubscriptionDeniedError|error? {
        // implement validation here
    }

    isolated remote function onSubscriptionIntentVerified(websubhub:VerifiedSubscription msg) returns error? {
        .// implement logic here
    }

    isolated remote function onUnsubscription(websubhub:Unsubscription msg, http:Headers headers)
               returns websubhub:UnsubscriptionAccepted|websubhub:BadUnsubscriptionError|websubhub:InternalUnsubscriptionError|error {
        // implement logic here
    }

    isolated remote function onUnsubscriptionValidation(websubhub:Unsubscription msg)
                returns websubhub:UnsubscriptionDeniedError|error? {
      // implement validation here
    }

    isolated remote function onUnsubscriptionIntentVerified(websubhub:VerifiedUnsubscription msg) returns error? {
    }
}
```
- [X] Fixes [#1477](https://github.com/ballerina-platform/ballerina-standard-library/issues/1477)

Additional non-remote methods should be allowed in `websubhub:Service` Declaration.
```ballerina
service /hub on new websubhub:Listener(9090) {
    remote function onRegisterTopic(websubhub:TopicRegistration message, http:Headers headers)
                                returns websubhub:TopicRegistrationSuccess|websubhub:TopicRegistrationError {
       // implement validation
       self.register(message);
    }

    // this should be valid
    function register(websubhub:TopicRegistration message) returns error? {
      // implement logic
    }
}
```
- [X] Part of [#1417](https://github.com/ballerina-platform/ballerina-standard-library/issues/1417)

## Checklist
- [X] Linked to an issue
- [X] Updated the changelog
- [X] Added tests

### CODE REVIEW

File: websubhub-compiler-plugin-test/src/test/resources/ballerina_sources/sample_17/service.bal [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                       isSuccess: "true"
+                    }
+            };
+        if (msg.hubTopic == "test") {
+            return successResult;
+        } else if (msg.hubTopic == "test1") {
+            return successResult;
+        } else {
+            return error websubhub:BadSubscriptionError("Bad subscription");
+        }
```
Comment (MadhukaHarith92) [inline websubhub-compiler-plugin-test/src/test/resources/ballerina_sources/sample_17/service.bal:67]:
```suggestion
        if (msg.hubTopic == "test" || msg.hubTopic == "test1") {
            return successResult;
        } else {
            return error websubhub:BadSubscriptionError("Bad subscription");
        }
```

File: websubhub-compiler-plugin/src/main/java/io/ballerina/stdlib/websubhub/task/validator/ServiceDeclarationValidator.java [hunk 2]
```diff
@@ -138,26 +159,28 @@ private void validateRequiredMethodsImplemented(SyntaxNodeAnalysisContext contex
...
-
-    private void validateAdditionalMethodImplemented(SyntaxNodeAnalysisContext context,
-                                                     FunctionDefinitionNode functionDefinition, NodeLocation location) {
-        String functionName = functionDefinition.functionName().toString();
-        if (!allowedMethods.contains(functionName)) {
-            WebSubHubDiagnosticCodes errorCode = WebSubHubDiagnosticCodes.WEBSUBHUB_104;
-            updateContext(context, errorCode, location, functionName);
-        }
-    }
+//    private void validateRemoteQualifier(SyntaxNodeAnalysisContext context, FunctionSymbol functionSymbol,
```
Comment (lahirusamith) [inline websubhub-compiler-plugin/src/main/java/io/ballerina/stdlib/websubhub/task/validator/ServiceDeclarationValidator.java:162]: Shall we remove the commented lines if those are not needed?

File: websubhub-compiler-plugin/src/main/java/io/ballerina/stdlib/websubhub/task/validator/ServiceDeclarationValidator.java [hunk 3]
```diff
@@ -110,21 +110,42 @@ public void validate(SyntaxNodeAnalysisContext context, ServiceDeclarationNode s
...
+        executeRequiredMethodValidation(context, availableFunctionDeclarations, serviceNode.location());
         availableFunctionDeclarations.forEach(fd -> {
             context.semanticModel().symbol(fd).ifPresent(fs -> {
                 NodeLocation location = fd.location();
-                validateRemoteQualifier(context, (FunctionSymbol) fs, location);
-                validateAdditionalMethodImplemented(context, fd, location);
-                validateMethodParameters(context, fd, ((FunctionSymbol) fs).typeDescriptor());
-                validateMethodReturnTypes(context, fd, ((FunctionSymbol) fs).typeDescriptor());
+                executeRemoteMethodValidation(context, (FunctionSymbol) fs, location);
+//                validateRemoteQualifier(context, (FunctionSymbol) fs, location);
```
Comment (lahirusamith) [inline websubhub-compiler-plugin/src/main/java/io/ballerina/stdlib/websubhub/task/validator/ServiceDeclarationValidator.java:118]: Is it okey to remove these commented lines?

**Comment by shafreenAnfar** [comment]
Does it includes a test with an extern function ?

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**